lib/credentials.js

```diff
@@ -131,6 +125,40 @@ Credentials.prototype.hasWalletInfo = function() {
   return !!this.walletId;
 };
 
+Credentials.prototype.isPrivKeyEncrypted = function() {
+  return (!!this.xPrivKeyEnc) && !this.xPrivKey;
+};
+
+Credentials.prototype.hasPrivKeyEncrypted = function() {
+  return (!!this.xPrivKeyEnc);
+};
+
+Credentials.prototype.encrypt = function(password, opts) {
+  if (this.xPrivKeyEnc)
+    throw new Error('Encrypted Privkey Already exists');
+
+  if (!this.xPrivKey)
+    throw new Error('No private key to encrypt');
+
+
+  this.xPrivKeyEnc = sjcl.encrypt(password, this.xPrivKey, opts);
+  if (!this.xPrivKeyEnc)
+    throw new Error('Could not encrypt');
+};
+
+Credentials.prototype.lock = function() {
```

Comment:
Why would we need two separate methods to encrypt? Wouldn't it be cleaner to both encrypt the pk and delete the unencrypted pk all at once?